kerastuner/engine/trial.py

```diff
@@ -17,34 +17,39 @@
 from __future__ import division
 from __future__ import print_function
 
-import os
+import hashlib
 import json
+import os
+import random
+import time
 
+from kerastuner.engine import stateful
 from . import metrics_tracking
 from . import hyperparameters as hp_module
-from . import execution as execution_module
 from ..abstractions import display
 from ..abstractions.tensorflow import TENSORFLOW_UTILS as tf_utils
 
 
-class Trial(object):
+class TrialStatus:
+    RUNNING = "RUNNING"
+    IDLE = "IDLE"
+    INVALID = "INVALID"
+    STOPPED = "STOPPED"
+    COMPLETED = "COMPLETED"
+
+
+class Trial(stateful.Stateful):
```

Comment:
This is the Oracle-side Trial, right? Do we have a Tuner-side Trial-like object as well? To keep track of directory and execution info

Reply:
Discussed offline, we should be able to serialize the Trial and handle executions in the MultiExecutionTuner